test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -51,11 +57,86 @@
+          jruby-service (reify jruby/JRubyPuppetService
+                          (borrow-instance [_ _] jruby-instance)
+                          (return-instance [_ jruby _]
+                            (reset! returned-jruby jruby))
+                          (get-environment-class-info [_ jruby env]
+                            (reset! get-env-jruby jruby)
+                            (if (= env "production")
+                              {"/one/file"
+                               [
+                                {"name" "oneclass",
+                                 "params" [
+                                           {"name" "oneparam",
+                                            "type" "String",
+                                            "default_literal" "'literal'",
+                                            "default_source" "literal"},
+                                           {"name" "twoparam",
+                                            "type" "Integer",
+                                            "default_literal" "3",
+                                            "default_source" "3"}]
+                                 },
+                                {"name" "twoclass"
+                                 "params" []}],
+                               "/two/file" []})))
+          handler (fn ([req] {:request req}))
+          app (build-ring-handler handler "1.2.3" jruby-service)
+          request (partial app-request app)]
+      (testing "returns good data for environment that exists"
+        (let [response (request
+                        "/v3/environment_classes?environment=production")
+              body-from-json (cheshire/parse-string (:body response))]
+          (is (= 200 (:status response))
+              "Unexpected status code returned from request")
+          (is (= {"name" "production",
+                  "files" [
+                           {"path" "/one/file",
+                            "classes" [{
+                                        "name" "oneclass"
+                                        "params" [
+                                                  {"name" "oneparam",
+                                                   "type" "String",
+                                                   "default_literal" "'literal'",
+                                                   "default_source" "literal"},
+                                                  {"name" "twoparam",
+                                                   "type" "Integer",
+                                                   "default_literal" "3",
+                                                   "default_source" "3"}]},
+                                       {
+                                        "name" "twoclass"
+                                        "params" []}]},
+                           {"path" "/two/file"
+                            "classes" []}]}
+                 body-from-json)
```

Comment:
I had been thinking that the tests at this level would be validating different good request / bad request behaviors whereas the ones at the integration level would be more about exercising the full path - from HTTP request down to Puppet Ruby code and back - for "success" cases but with a little more emphasis on the variations of the return payload to ensure for proper translation across the full path, from Ruby -> Clojure -> Serialized JSON -> Network -> Deserialized JSON.

The "returns good date for environment that exists" test block here is arguably redundant with what gets covered at the integration layer, I agree.  I wrote this test block first before everything else was working so it had value initially.  I suppose I could keep the check for "200 OK" in this test block but not do any of the other checks - JRuby borrowing behaviors and that the actual body of the JSON payload is handled properly - at this level.  Dunno.  WDYT?